src/main/java/ome/formats/importer/targets/ModelImportTarget.java

```diff
@@ -131,7 +130,13 @@ public IObject load(OMEROMetadataStoreClient client, ImportContainer ic) throws
                 "select o from "+simpleName+" as o where o.name = :name"
                 + " order by o.id " + order,
                 new ParametersI().add("name", rstring(name)));
-            if (objs.size() == 0 || discriminator.startsWith("@")) {
+            final Iterator<IObject> objIter = objs.iterator();
+            while (objIter.hasNext()) {
+                if (!objIter.next().getDetails().getPermissions().canLink()) {
+                    objIter.remove();
+                }
+            }
+            if (objs.isEmpty() || discriminator.startsWith("@")) {
```

Comment:
Perhaps skip the query completely if the discriminator is `@`?

Reply:
I had thought about that. :smiley: I was going for a minimal change but can certainly be more invasive.